### PR TITLE
Add bms in bmpclient and command

### PR DESCRIPTION
**Pull request by zhanggbj**
Hi,
I added missing bms in bmpclient, bms command and their UT. Thanks for your review!

### CODE REVIEW

File: clients/bmp_client_test.go [hunk 1]
```diff
@@ -62,6 +62,47 @@ var _ = Describe("BMP client", func() {
...
+			Expect(len(bmsResponse.Data)).To(Equal(2))
+			Expect(bmsResponse.Data[0]).To(Equal(clients.BaremetalInfo{
+				Id:   0,
+				Hostname: "hostname0",
+				Private_ip_address: "private_ip_address0",
+				Public_ip_address: "public_ip_address0",
+				Hardware_status: "hardware_status0",
+				Memory: 0,
+				Cpu: 0,
+				Provision_date: "provision_date0"}))
```
Comment (maximilien) [inline clients/bmp_client_test.go]: This probably has to be a be a valid date (string encoded)

File: clients/data_types.go [hunk 2]
```diff
@@ -131,3 +148,8 @@ type CreateBaremetalInfo struct {
 	BaremetalSpecs []CloudProperty `json:"baremetal_specs"`
 	Deployment     string          `json:"deployment"`
 }
+
+// deployment
+type Deployment struct {
+	Name string `yaml:"name"`
```
Comment (maximilien) [inline clients/data_types.go]: Yaml? Please explain

File: cmds/bmp/bms_command.go [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+	"io/ioutil"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
 	clients "github.com/cloudfoundry-community/bosh-softlayer-tools/clients"
 	cmds "github.com/cloudfoundry-community/bosh-softlayer-tools/cmds"
 	common "github.com/cloudfoundry-community/bosh-softlayer-tools/common"
+	tablewriter "github.com/olekukonko/tablewriter"
```
Comment (maximilien) [inline cmds/bmp/bms_command.go:16]: DK also has a table writer in golang. Might want to ask him. I don't know this one so before we commit ask @cppforlife for his, maybe he knows how it compares to this one...

File: cmds/bmp/bms_command_test.go [hunk 4]
```diff
@@ -70,13 +75,69 @@ var _ = Describe("bms command", func() {
...
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		Context("bad BmsCommand", func() {
+			Context("no deployment file", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose: false,
+						Deployment: "",
+					}
```
Comment (maximilien) [inline cmds/bmp/bms_command_test.go]: Use fake-filename here or similar

Comment (maximilien) [inline cmds/bmp/bms_command_test.go]: Unless you are specifically testing empty string

File: cmds/bmp/bms_command_test.go [hunk 5]
```diff
@@ -70,13 +75,69 @@ var _ = Describe("bms command", func() {
...
-			Expect(rc).To(Equal(0))
-			Expect(err).ToNot(HaveOccurred())
+		Context("executes a good BmsCommand", func() {
+			BeforeEach(func() {
+				fakeBmpClient.BmsResponse.Status = 200
+				fakeBmpClient.BmsErr = nil
+			})
+
+			It("executes with no error", func() {
+				rc, err := cmd.Execute(args)
```
Comment (maximilien) [inline cmds/bmp/bms_command_test.go]: What are the args here?

File: test_fixtures/bmp/Bms.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    {
+      "id":   1,
+      "hostname": "hostname1",
+      "private_ip_address": "private_ip_address1",
+      "public_ip_address": "public_ip_address1",
+      "hardware_status": "hardware_status1",
+      "memory": 1,
+      "cpu": 1,
+      "provision_date": "provision_date1"
```
Comment (maximilien) [inline test_fixtures/bmp/Bms.json]: Same as before. This needs to be a valid string date

**Comment by maximilien** [comment]
Looks good but please address comments and I'll review again and maybe merge

File: test_fixtures/bmp/Bms.json (hunk 6 above)
Comment (zhanggbj) [inline test_fixtures/bmp/Bms.json]: Done

File: cmds/bmp/bms_command_test.go (hunk 5 above)
Comment (zhanggbj) [inline cmds/bmp/bms_command_test.go]: args here are "bmp bms"

File: cmds/bmp/bms_command.go (hunk 3 above)
Comment (zhanggbj) [inline cmds/bmp/bms_command.go:16]: Hi Dmitriy @cppforlife，
Max mentioned you have a table writer, too. Could you please share the link with me and I would like to have a try:)
Thanks in advance!

File: clients/data_types.go (hunk 2 above)
Comment (zhanggbj) [inline clients/data_types.go]: The Deployment type is used when to unmarshal the yaml file, so I specified it with 'yaml'.(seems 'json' also works here)

File: clients/bmp_client_test.go (hunk 1 above)
Comment (zhanggbj) [inline clients/bmp_client_test.go]: done.

File: cmds/bmp/bms_command_test.go (hunk 4 above)
Comment (zhanggbj) [inline cmds/bmp/bms_command_test.go]: My intention is to test empty string, but after a second thought, I think it is not so meaningful here, so removed it. (The next case will verify fake file name)

File: clients/data_types.go (hunk 2 above)
Comment (maximilien) [inline clients/data_types.go]: yeah, change to json, you are not expecting YAML content which is what using yaml will do.

File: cmds/bmp/bms_command.go (hunk 3 above)
Comment (maximilien) [inline cmds/bmp/bms_command.go:16]: I'll merge with what you have now but if the @cppforlife is better then let's look into switching to it as a chore. Especially since I know DK's is being used or will be used in BOSH code and might have coloring which would be nice.

File: cmds/bmp/bms_command_test.go (hunk 5 above)
Comment (maximilien) [inline cmds/bmp/bms_command_test.go]: I would rather see this be more explicit, e.g., `cmd.Execute([]string{"bms", "bms"})` or similar.

File: cmds/bmp/login_command_test.go [hunk 7]
```diff
@@ -72,7 +72,7 @@ var _ = Describe("login command", func() {
 
 	Describe("#Usage", func() {
 		It("returns the usage text of a LoginCommand", func() {
-			Expect(cmd.Usage()).To(Equal("bmp login --username[-u] <username> --password[-p] <password"))
+			Expect(cmd.Usage()).To(Equal("bmp login --username[-u] <username> --password[-p] <password>"))
 		})
 	})
```
Comment (maximilien) [inline cmds/bmp/login_command_test.go:77]: these login changes should be moved to their own PR. Let's not mix things, unless there is an explicit dependency(ies)

**Comment by maximilien** [comment]
OK, last set of comments. Please address and we should be ready to merge. Thanks.

File: cmds/bmp/login_command_test.go (hunk 7 above)
Comment (zhanggbj) [inline cmds/bmp/login_command_test.go:77]: I see.  Will be more careful next time.

File: clients/data_types.go (hunk 2 above)
Comment (zhanggbj) [inline clients/data_types.go]: done

File: cmds/bmp/bms_command_test.go (hunk 5 above)
Comment (zhanggbj) [inline cmds/bmp/bms_command_test.go]: done

**Comment by zhanggbj** [comment]
Hi @maximilien ,

Here is the update. Thank you for your review:)

**Comment by maximilien** [comment]
Looks good. Merging 🍻
